File: public/classes/580.s16/code/A1Encr.java

public class A1Encr {
    static String plaintext = "OMITTED!";

    public static void main(String argv[]) {
	int ka = 0, kb = 0;  // These two values hold the secret key

	for (int i=0; i<plaintext.length(); i++) {
	    int ctval = (ka * plaintext.charAt(i) + kb)%256;
	    System.out.printf("\\%o",ctval);
	}
    }
}

File: public/classes/580.s16/code/A1Encr.cpp
#include <iostream>
#include <string>

using namespace std;

string plaintext = "OMITTED!";

int main(int argc, char *argv[])
{
    int ka = 0, kb = 0;  // These two values hold the secret key

    for (unsigned int i=0; i<plaintext.length(); i++) {
	int ctval = (ka * (unsigned char)plaintext[i] + kb)%256;
	cout << "\\" << oct << ctval;
    }
}
